base/abstractarray.jl

```diff
@@ -2112,14 +2112,34 @@ _hvncat(dimsshape::Union{Tuple, Int}, row_first::Bool, xs::AbstractArray...) = _
 _hvncat(dimsshape::Union{Tuple, Int}, row_first::Bool, xs::AbstractArray{T}...) where T = _typed_hvncat(T, dimsshape, row_first, xs...)
 
 typed_hvncat(::Type{T}, ::Tuple{}, ::Bool) where T = Vector{T}()
-typed_hvncat(::Type{T}, ::Tuple{}, ::Bool, xs...) where T = Vector{T}()
-typed_hvncat(T::Type, ::Tuple{Vararg{Any, 1}}, ::Bool, xs...) = typed_vcat(T, xs...) # methods assume 2+ dimensions
 typed_hvncat(T::Type, dimsshape::Tuple, row_first::Bool, xs...) = _typed_hvncat(T, dimsshape, row_first, xs...)
 typed_hvncat(T::Type, dim::Int, xs...) = _typed_hvncat(T, Val(dim), xs...)
 
+# 1-dimensional hvncat methods
+
+_typed_hvncat(::Type{T}, ::Val{0}, x) where T = fill(T(x))
+_typed_hvncat(::Type{T}, ::Val{0}, x::Number) where T = fill(T(x))
```

Comment:
This shouldn't be needed right?

Reply:
Julia says it's ambiguous if I remove it. Same for the rest.

Reply:
Ah, I see, didn't think of that. It's a bit unfortunate that we have to duplicate this, but seems reasonable then.